packages/runtime/container-runtime/src/gc/gcConfigs.ts

```diff
@@ -158,9 +158,7 @@ export function generateGCConfigs(
 
 	// If version upgrade is not enabled, fall back to the stable GC version.
 	const gcVersionInEffect =
-		mc.config.getBoolean(gcVersionUpgradeToV3Key) === false
-			? stableGCVersion
-			: currentGCVersion;
+		mc.config.getBoolean(gcVersionUpgradeToV3Key) === true ? currentGCVersion : stableGCVersion;
```

Comment:
If you have to make any other changes, I'd rename `current` to `next`, but it's not worth wasting the passed checks to change it IMO.